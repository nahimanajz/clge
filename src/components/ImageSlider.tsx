"use client";

import "react-responsive-carousel/lib/styles/carousel.min.css";
import { Carousel } from "react-responsive-carousel";
import { FC } from "react";
import { IImageSliderProps } from "@/shared/props";
import Image from "next/image";

const ImageSlider: FC<IImageSliderProps> = ({ images }) => {
  //TODO: REMOVE DUPLICATED IMAGES  AND DEFINE ONE IMAGE WITH PROPS
  return (
   
      <Carousel showArrows={true} showThumbs={false}>
        <Image
          src={`${images?.back}`}
          alt="back"
          className="rounded-[8px]"
          width={300}
          height={300}
        />
        <Image
          src={`${images?.front}`}
          alt="front"
          className="rounded-[8px]"
          width={300}
          height={300}
        />
        <Image
          src={`${images?.leftSide}`}
          alt="left"
          className="rounded-[8px]"
          width={300}
          height={300}
        />
        <Image
          src={`${images?.rightSide}`}
          alt="right"
          className="rounded-[8px]"
          width={300}
          height={300}
        />
      </Carousel>
   
  );
};
export default ImageSlider;
