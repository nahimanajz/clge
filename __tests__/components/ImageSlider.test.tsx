import { render } from "@testing-library/react";
import ImageSlider from "@/components/ImageSlider";

describe("room card test ", () => {
  const images = {
    id: "1",
    leftSide:
      "https://www.thespruce.com/thmb/iMt63n8NGCojUETr6-T8oj-5-ns=/1500x0/filters:no_upscale():max_bytes(150000):strip_icc()/PAinteriors-7-cafe9c2bd6be4823b9345e591e4f367f.jpg",
    rightSide:
      "https://www.thespruce.com/thmb/iMt63n8NGCojUETr6-T8oj-5-ns=/1500x0/filters:no_upscale():max_bytes(150000):strip_icc()/PAinteriors-7-cafe9c2bd6be4823b9345e591e4f367f.jpg",
    front:
      "https://www.thespruce.com/thmb/iMt63n8NGCojUETr6-T8oj-5-ns=/1500x0/filters:no_upscale():max_bytes(150000):strip_icc()/PAinteriors-7-cafe9c2bd6be4823b9345e591e4f367f.jpg",
    back: "https://www.thespruce.com/thmb/iMt63n8NGCojUETr6-T8oj-5-ns=/1500x0/filters:no_upscale():max_bytes(150000):strip_icc()/PAinteriors-7-cafe9c2bd6be4823b9345e591e4f367f.jpg",
  };

  it(" should match slider snapshot", () => {
    const { container } = render(<ImageSlider images={images} />);
    expect(container).toMatchSnapshot();
  });
});
