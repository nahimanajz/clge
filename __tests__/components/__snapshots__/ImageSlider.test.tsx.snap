// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`room card test   should match slider snapshot 1`] = `
<div>
  <div
    class="carousel-root"
  >
    <div
      class="carousel carousel-slider"
      style="width: 100%;"
    >
      <ul
        class="control-dots"
      >
        <li
          aria-label="slide item 1"
          class="dot selected"
          role="button"
          tabindex="0"
          value="0"
        />
        <li
          aria-label="slide item 2"
          class="dot"
          role="button"
          tabindex="0"
          value="1"
        />
        <li
          aria-label="slide item 3"
          class="dot"
          role="button"
          tabindex="0"
          value="2"
        />
        <li
          aria-label="slide item 4"
          class="dot"
          role="button"
          tabindex="0"
          value="3"
        />
      </ul>
      <button
        aria-label="previous slide / item"
        class="control-arrow control-prev control-disabled"
        type="button"
      />
      <div
        class="slider-wrapper axis-horizontal"
      >
        <ul
          class="slider animated"
          style="transform: translate3d(0,0,0); transition-duration: 350ms;"
        >
          <li
            class="slide selected previous"
          >
            <img
              alt="back"
              class="rounded-[8px]"
              data-nimg="1"
              decoding="async"
              height="300"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Fwww.thespruce.com%2Fthmb%2FiMt63n8NGCojUETr6-T8oj-5-ns%3D%2F1500x0%2Ffilters%3Ano_upscale()%3Amax_bytes(150000)%3Astrip_icc()%2FPAinteriors-7-cafe9c2bd6be4823b9345e591e4f367f.jpg&w=640&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fwww.thespruce.com%2Fthmb%2FiMt63n8NGCojUETr6-T8oj-5-ns%3D%2F1500x0%2Ffilters%3Ano_upscale()%3Amax_bytes(150000)%3Astrip_icc()%2FPAinteriors-7-cafe9c2bd6be4823b9345e591e4f367f.jpg&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fwww.thespruce.com%2Fthmb%2FiMt63n8NGCojUETr6-T8oj-5-ns%3D%2F1500x0%2Ffilters%3Ano_upscale()%3Amax_bytes(150000)%3Astrip_icc()%2FPAinteriors-7-cafe9c2bd6be4823b9345e591e4f367f.jpg&w=640&q=75 2x"
              style="color: transparent;"
              width="300"
            />
          </li>
          <li
            class="slide"
          >
            <img
              alt="front"
              class="rounded-[8px]"
              data-nimg="1"
              decoding="async"
              height="300"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Fwww.thespruce.com%2Fthmb%2FiMt63n8NGCojUETr6-T8oj-5-ns%3D%2F1500x0%2Ffilters%3Ano_upscale()%3Amax_bytes(150000)%3Astrip_icc()%2FPAinteriors-7-cafe9c2bd6be4823b9345e591e4f367f.jpg&w=640&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fwww.thespruce.com%2Fthmb%2FiMt63n8NGCojUETr6-T8oj-5-ns%3D%2F1500x0%2Ffilters%3Ano_upscale()%3Amax_bytes(150000)%3Astrip_icc()%2FPAinteriors-7-cafe9c2bd6be4823b9345e591e4f367f.jpg&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fwww.thespruce.com%2Fthmb%2FiMt63n8NGCojUETr6-T8oj-5-ns%3D%2F1500x0%2Ffilters%3Ano_upscale()%3Amax_bytes(150000)%3Astrip_icc()%2FPAinteriors-7-cafe9c2bd6be4823b9345e591e4f367f.jpg&w=640&q=75 2x"
              style="color: transparent;"
              width="300"
            />
          </li>
          <li
            class="slide"
          >
            <img
              alt="left"
              class="rounded-[8px]"
              data-nimg="1"
              decoding="async"
              height="300"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Fwww.thespruce.com%2Fthmb%2FiMt63n8NGCojUETr6-T8oj-5-ns%3D%2F1500x0%2Ffilters%3Ano_upscale()%3Amax_bytes(150000)%3Astrip_icc()%2FPAinteriors-7-cafe9c2bd6be4823b9345e591e4f367f.jpg&w=640&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fwww.thespruce.com%2Fthmb%2FiMt63n8NGCojUETr6-T8oj-5-ns%3D%2F1500x0%2Ffilters%3Ano_upscale()%3Amax_bytes(150000)%3Astrip_icc()%2FPAinteriors-7-cafe9c2bd6be4823b9345e591e4f367f.jpg&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fwww.thespruce.com%2Fthmb%2FiMt63n8NGCojUETr6-T8oj-5-ns%3D%2F1500x0%2Ffilters%3Ano_upscale()%3Amax_bytes(150000)%3Astrip_icc()%2FPAinteriors-7-cafe9c2bd6be4823b9345e591e4f367f.jpg&w=640&q=75 2x"
              style="color: transparent;"
              width="300"
            />
          </li>
          <li
            class="slide"
          >
            <img
              alt="right"
              class="rounded-[8px]"
              data-nimg="1"
              decoding="async"
              height="300"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Fwww.thespruce.com%2Fthmb%2FiMt63n8NGCojUETr6-T8oj-5-ns%3D%2F1500x0%2Ffilters%3Ano_upscale()%3Amax_bytes(150000)%3Astrip_icc()%2FPAinteriors-7-cafe9c2bd6be4823b9345e591e4f367f.jpg&w=640&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fwww.thespruce.com%2Fthmb%2FiMt63n8NGCojUETr6-T8oj-5-ns%3D%2F1500x0%2Ffilters%3Ano_upscale()%3Amax_bytes(150000)%3Astrip_icc()%2FPAinteriors-7-cafe9c2bd6be4823b9345e591e4f367f.jpg&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fwww.thespruce.com%2Fthmb%2FiMt63n8NGCojUETr6-T8oj-5-ns%3D%2F1500x0%2Ffilters%3Ano_upscale()%3Amax_bytes(150000)%3Astrip_icc()%2FPAinteriors-7-cafe9c2bd6be4823b9345e591e4f367f.jpg&w=640&q=75 2x"
              style="color: transparent;"
              width="300"
            />
          </li>
        </ul>
      </div>
      <button
        aria-label="next slide / item"
        class="control-arrow control-next"
        type="button"
      />
      <p
        class="carousel-status"
      >
        1 of 4
      </p>
    </div>
  </div>
</div>
`;
